### ISSUE TITLE
Add related categories to "More Info" section on add-on details

### ISSUE THREAD

**Issue by tofumatt** [mozilla/addons-frontend#2758]
Add related categories to "More Info" section on add-on details. Need to double-check if this has UX mocks, I don't recall them...

**Comment by muffinresearch** [comment]
Currently the data only has the slug.

**Comment by muffinresearch** [comment]
mozilla/addons-server#6422 was wont-fixed so we will be needing to make an additional call to get the categories for the add-on.

**Comment by muffinresearch** [comment]
@pwalm it doesn't look like add-on categories are currently represented in the more information panel

<img width="302" alt="amo_desktop_-_master_sketch" src="https://user-images.githubusercontent.com/1514/32047918-6b692a3c-ba3f-11e7-9a5d-16a6b6a44bec.png">

**Comment by willdurand** [comment]
@muffinresearch is it still something we want to implement?

**Comment by jvillalobos** [comment]
Yes, we still want categories to be linked from the listing.

**Comment by muskangupta-iitr** [comment]
@willdurand @jvillalobos Can I take up this issue for adding related categories in `More Information` section of add-ons ?

**Comment by willdurand** [comment]
> @willdurand @jvillalobos Can I take up this issue for adding related categories in `More Information` section of add-ons ?

Yep

**Comment by muskangupta-iitr** [comment]
@willdurand Just want to ask few doubts: 
For example: Categories for `DuckDuckGo Plus` looks like:
```
categories: {
       android: [
            'feeds-news-blogging',
            'security-privacy'
       ],
       firefox: [
            'privacy-security',
            'search-tools'
       ]
},
```
Do we need to show both categories or check the clientApp whether it is `firefox` or `android` and then list categories accordingly ?
For UI, I am thinking to show categories as same as in the [extension section](https://addons.mozilla.org/en-US/firefox/extensions/). Any comments ?

**Comment by jvillalobos** [comment]
> Just want to confirm that I have to first check the clientApp whether it is firefox or android and then list categories accordingly ?

Yes.

> For UI, I am thinking to show categories as same as in the extension section. Any comments ?

See the mockup in [this comment](https://github.com/mozilla/addons-frontend/issues/2758#issuecomment-339620743) for how it should look.

**Comment by muskangupta-iitr** [comment]
> ```
> categories: {
>        android: [
>             'feeds-news-blogging',
>             'security-privacy'
>        ],
>        firefox: [
>             'privacy-security',
>             'search-tools'
>        ]
> },
> ```
> 
> And only category slug is coming in the object, name is not there ?

Any suggestion on this?

**Comment by muskangupta-iitr** [comment]
@willdurand @jvillalobos I have created a draft PR becuase currently category name is not coming with the object. Please review the current changes and how we further proceed on this ?

**Comment by willdurand** [comment]
> @willdurand @jvillalobos I have created a draft PR becuase currently category name is not coming with the object. Please review the current changes and how should we further proceed on this ?

Yeah, we'll have to look into the `categories` reducer to see if we have the categories loaded (and fetch them otherwise). This is going to be complex :)

**Comment by muskangupta-iitr** [comment]
@willdurand Can we fetch a category details individually using its slug? or Can we fetch all categories related to a addon separately?

**Comment by willdurand** [comment]
@muskangupta-iitr so we can't fetch one category but we can fetch all the categories (once!) and then use them. This is how we do it for category pages: https://github.com/xlisachan/addons-frontend/blob/4c604542c6f8d785bfe7586df3d677f0355584fd/src/amo/components/Categories/index.js#L87

We will have to make sure that we only fetch the categories if we need to. There is a `categories` reducer that stores the categories by `appName` (so `firefox` or `android`) and then by `addonType` (`extension`, `theme`, etc.).

**Comment by muskangupta-iitr** [comment]
Ohk @willdurand I will update the PR with the suitable changes. Thanks :+1: 
But why can't we add a API for fetching a single category as it might be used in future or we can serialize the addon category with whole category serializer?

**Comment by muskangupta-iitr** [comment]
@willdurand For now I pushed a commit in which I am fetching categories in `AddonMoreInfo` component and it's working fine. But I am facing problems in writing its tests for categories.

**Comment by bobsilverberg** [comment]
@muskangupta-iitr I can help you with this. I'll take a look at the PR later and we can discuss it there.

**Comment by muskangupta-iitr** [comment]
@bobsilverberg Thanks! I will let you know once I got stuck anywhere.

**Comment by bobsilverberg** [comment]
Just a note: A lot of work was done towards this in https://github.com/mozilla/addons-frontend/pull/9253, which was unfortunately abandoned by the contributor. When this is next taken up by someone, it probably makes sense to use the code from that PR as a starting point, as it has already been heavily reviewed.

**Comment by xlisachan** [comment]
@bobsilverberg , can I work on this?

**Comment by bobsilverberg** [comment]
@xlisachan It looks like @eviljeff just added this to a project that he is working on for Q3, so I'm going to suggest you discuss it with him, in case it's something that he prefers to take on himself.

**Comment by eviljeff** [comment]
@xlisachan you can go ahead and work on it - I just added it to the project because Jorge mentioned it as something he'd like to be implemented while we're doing something similar with exposing tags in the add-on details section.  But it's not connected to that work directly (and I appreciate it being taken off my plate 😄)

**Comment by ioanarusiczki** [comment]
@willdurand @xlisachan 

- "Related categories" is present into "More information" card for extensions and static themes. Could not find UX issues while resizing the detail pages.
- as functionality, I think it works as expected, I chose 2 categories from dev hub, changed them and then checked if the URLs redirect correctly to those categories. I did not run into errors.  Extensions can display 2 categories, themes have one option.
- looks good on Android too.

 
![categories](https://user-images.githubusercontent.com/33448286/129029047-35a740cb-38da-4a93-a10e-d15bdde00017.gif)
